apps/meteor/server/models/raw/Messages.ts

```diff
@@ -243,6 +253,70 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 		);
 	}
 
+	findVisibleByRoomIdNotContainingTypesBeforeTs(
+		roomId: IRoom['_id'],
+		types: IMessage['t'][],
+		ts: Date,
+		options?: FindOptions<IMessage>,
+		showThreadMessages = true,
+	): FindCursor<IMessage> {
+		const query: Filter<IMessage> = {
+			_hidden: {
+				$ne: true,
+			},
+			rid: roomId,
+			ts: { $lt: ts },
+			...(!showThreadMessages && {
+				$or: [
+					{
+						tmid: { $exists: false },
+					},
+					{
+						tshow: true,
+					},
+				],
+			}),
+		};
+
+		if (types.length > 0) {
+			query.t = { $nin: types };
+		}
+
+		return this.find(query, options);
+	}
+
+	findVisibleByRoomIdNotContainingTypesAndUsers(
```

Comment:
Do we still need this method?

Reply:
Not for this PR, but I have other place where i need this same thing so i'll leave it for now :awesome: